core/src/main/java/com/web3auth/core/Web3Auth.kt

```diff
@@ -168,30 +155,31 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
      *
      * @return A CompletableFuture<Void> representing the asynchronous operation.
      */
-    fun initialize(): CompletableFuture<Void> {
+    fun initialize(context: Context): CompletableFuture<Void> {
         val initializeCf = CompletableFuture<Void>()
-        KeyStoreManagerUtils.initializePreferences(web3AuthOption.context.applicationContext)
+        KeyStoreManagerUtils.initializePreferences(context.applicationContext)
 
         //initiate keyStore
         initiateKeyStoreManager()
 
-        //authorize session
-        if (ApiHelper.isNetworkAvailable(web3AuthOption.context)) {
-
-            //fetch project config
-            fetchProjectConfig().whenComplete { _, err ->
-                if (err == null) {
-                    this.authorizeSession().whenComplete { resp, error ->
-                        if (error == null) {
-                            web3AuthResponse = resp
-                        } else {
-                            print(error)
+        //fetch project config
+        fetchProjectConfig(context).whenComplete { _, err ->
+            if (err == null) {
+                //authorize session
+                sessionManager.setSessionId(SessionManager.getSessionIdFromStorage())
+                this.authorizeSession(web3AuthOption.redirectUrl.toString(), context)
+                    .whenComplete { resp, error ->
+                        runOnUIThread {
+                            if (error == null) {
+                                web3AuthResponse = resp
+                            } else {
+                                initializeCf.completeExceptionally(error)
+                            }
+                            initializeCf.complete(null)
```

Comment:
Is it fine to call both
`initializeCf.completeExceptionally(error)`
and 
`initializeCf.complete(null)`
right after each other in case of error?

Reply:
But what's the need of it? Since **.completeExceptionally** completes the future with exception and **.complete** completes the future with any <T> result. So ideally wither of them should be called.

Reply:
Should not make a difference.

According to the docs it is conditional on it not already being completed for both functions:
https://docs.oracle.com/javase/8/docs/api/index.html?java/util/concurrent/CompletableFuture.html

Reply:
@grvgoel81 For clarity, it may be better to move line 178 to just after line 174 here.